api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -88,6 +88,15 @@
   String USE_STARTING_SEQUENCE_NUMBER = "use-starting-sequence-number";
   boolean USE_STARTING_SEQUENCE_NUMBER_DEFAULT = true;
 
+  /**
+   * Forces the compaction order according to the partition size in decreasing order, instead of
```

Comment:
This is a bit internal, but do we mean per fileGroup here as well? For example if a partition has 2 file groups, 100GB and 2 gb, do they both go before a partition with 1 file group of (50 gb)?

Reply:
I would let the users control the comparator by two options: 
- `rewrite-partitions-by-bytes` when enabled by the user should sort the partitions by bytes (higher to lower).
- `rewrite-partitions-by-num-files` when enabled by the user should sort the partitions by number of files in it (higher to lower).
- If nothing is enabled use the natural partition order.
- User cannot enable both the options.

In the above example, 

If `rewrite-partitions-by-bytes` is enabled then file group 1 (100 GB + 2GB = 102 GB) should go before file group 2 (50 GB). 
If `rewrite-partitions-by-num-files` the file group having more files should go before.

Reply:
Don't you think it should be ascending size of bytes? Since the smaller jobs are more likely to finish quickly and allow for interleaving of more jobs and the larger ones are more likely to be stragglers?

I think also maybe it should be something like

job-order: {bytes | files | none/unset }

Reply:
I see, I have made the changes to introduce `rewrite.job-order = {"bytes", "files", "none"}`.

* If `rewrite.job-order=bytes`, then compact based on ascending order of partition size in bytes.
* If `rewrite.job-order=files`, then compact based on descending order of number of files.
* If `rewrite.job-order=none`, then compact based on partition order.

Do you feel if there is a use case for descending order for partition size in bytes / ascending order of number of files? Or maybe some custom option which lets the user to pass any comparator as per their choice?